### PR TITLE
TOOLS/PERF: Add user-side memcpy option for AM benchmarks

### PR DESCRIPTION
**Pull request by ivankochin**
## What
Add option to perform additional user-side memcpy for AM benchmarks.

### CODE REVIEW

File: src/tools/perf/api/libperf.h [hunk 1]
```diff
@@ -88,7 +88,8 @@ enum ucx_perf_test_flags {
     UCX_PERF_TEST_FLAG_WAKEUP           = UCS_BIT(10), /* Create context with wakeup feature enabled */
     UCX_PERF_TEST_FLAG_ERR_HANDLING     = UCS_BIT(11), /* Create UCP eps with error handling support */
     UCX_PERF_TEST_FLAG_LOOPBACK         = UCS_BIT(12), /* Use loopback connection */
-    UCX_PERF_TEST_FLAG_PREREG           = UCS_BIT(13) /* Pass pre-registered memory handle */
+    UCX_PERF_TEST_FLAG_PREREG           = UCS_BIT(13), /* Pass pre-registered memory handle */
+    UCX_PERF_TEST_FLAG_AM_MEMCPY        = UCS_BIT(14) /* Do additional memcopy during AM receive */
```
Comment (yosefe) [inline src/tools/perf/api/libperf.h:92]: maybe UCX_PERF_TEST_FLAG_AM_RECV_COPY?
also pls align the comment on column

File: src/tools/perf/lib/ucp_tests.cc [hunk 2]
```diff
@@ -315,7 +315,11 @@ class ucp_perf_test_runner {
             return test->am_rndv_recv(data, length, param);
         }
 
-        /* TODO: Add option to do memcopy here */
+        if (test->m_perf.params.flags & UCX_PERF_TEST_FLAG_AM_MEMCPY) {
+            ucs_assert(length == test->m_am_rx_length);
```
Comment (yosefe) [inline src/tools/perf/lib/ucp_tests.cc:319]: assertv

File: test/gtest/ucp/test_ucp_perf.cc [hunk 3]
```diff
@@ -273,6 +273,13 @@ const test_perf::test_spec test_ucp_perf::tests[] =
     ucs_offsetof(ucx_perf_result_t, latency.total_average), 1e6, 0.001, 60.0,
     0 },
 
+  { "am_lat_memcpy", "usec",
+    UCX_PERF_API_UCP, UCX_PERF_CMD_AM, UCX_PERF_TEST_TYPE_PINGPONG,
+    UCX_PERF_WAIT_MODE_POLL,
+    UCP_PERF_DATATYPE_CONTIG, 0, 1, { 8 }, 1, 100000lu,
+    ucs_offsetof(ucx_perf_result_t, latency.total_average), 1e6, 0.001, 60.0,
+    UCX_PERF_TEST_FLAG_AM_MEMCPY },
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_perf.cc:281]: 👍

File: src/tools/perf/perftest_params.c [hunk 4]
```diff
@@ -134,6 +134,7 @@ static void usage(const struct perftest_context *ctx, const char *program)
     printf("                        sleep      : go to sleep after posting requests\n");
     printf("     -H <size>      active message header size (%zu), not included in message size\n",
                                 ctx->params.super.ucp.am_hdr_size);
+    printf("     -y             do additional memcopy to the user memory during active message receive handler\n");
```
Comment (brminich) [inline src/tools/perf/perftest_params.c:137]: minor: `..in active message receive handler`?

File: src/tools/perf/lib/ucp_tests.cc [hunk 5]
```diff
@@ -315,8 +315,8 @@ class ucp_perf_test_runner {
             return test->am_rndv_recv(data, length, param);
         }
 
-        if (test->m_perf.params.flags & UCX_PERF_TEST_FLAG_AM_MEMCPY) {
-            ucs_assert(length == test->m_am_rx_length);
+        if (test->m_perf.params.flags & UCX_PERF_TEST_FLAG_AM_RECV_COPY) {
+            ucs_assertv(length == test->m_am_rx_length, "wrong buffer length");
```
Comment (brminich) [inline src/tools/perf/lib/ucp_tests.cc:319]: pls print also both lengths

**Review by yosefe — DISMISSED** [review]

**Review by yosefe — APPROVED** [review]

**Pull request merged**